src/plugins/data_table/filter_view.js

```diff
@@ -68,43 +76,79 @@ var DataTableFilter = (function(options) {
         this.filter.value = filterInput.value;
         this.parentView._toggleIcon(true);
       }
+    },
+
+    afterRender: function() {
+      this.$('.btn-filter').click(this.parentView._onFilterClick.bind(this.parentView));
+      this.$('.btn-clear').click(this.parentView._onClearClick.bind(this.parentView));
     }
   });
 
   var NumericFilterMenu = DataTableFilterMenu.extend({
-    tagName: "ul",
-
     menuTemplate: _.template('\
-        <li> &gt; <input data-numeric-filter-name="gt" class="filter-numeric filter-numeric-greater" type="text" /></li> \
-        <li> &lt; <input data-numeric-filter-name="lt" class="filter-numeric filter-numeric-less" type="text"/></li> \
-        <li> = <input data-numeric-filter-name="eq" class="filter-numeric filter-numeric-equal" type="text" /></li> \
-      ', null, DEFAULT_JST_DELIMS),
+      <div class="filterMenu filterMenu-numeric">\
+        <div class="filter-text">Show items with value that:</div>\
+        <select class="filter-type form-control" data-filter-id="first-filter">\
+          <option selected value="gt">is greater than</option> \
+          <option value="lt">is less than</option> \
+          <option value="eq">is equal to</option> \
+        </select> \
+        <input id="first-filter" class="filter-value form-control" type="text" data-filter-type="gt"  /> \
+        <div class="filter-text">and</div> \
+        <select class="filter-type form-control" data-filter-id="second-filter">\
+          <option value="gt">is greater than</option> \
+          <option selected value="lt">is less than</option> \
+          <option value="eq">is equal to</option> \
+        </select> \
+        <input id="second-filter" class="filter-value form-control" type="text" data-filter-type="lt" /> \
+        <button class="btn btn-sm btn-filter" name="button" type="submit" title="">Apply</button>\
+        <button class="btn btn-primary btn-sm btn-clear pull-right" name="button" type="submit" title="">Clear</button>\
+      </div>\
+    ', null, DEFAULT_JST_DELIMS),
 
     _onInputChange: function (event) {
-      var filterInput = event.target;
-      var numericValueName = $(filterInput).attr("data-numeric-filter-name");
-      if (filterInput.value === "") {
-        this.filter[numericValueName] = null;
+      event.stopImmediatePropagation();
```

Comment:
If we don't stop propagation here, it triggers the sorting when you click the filter menu toggle button.